Realm/RLMRealmConfiguration+Sync.mm

```diff
@@ -45,18 +45,16 @@ - (void)setSyncConfiguration:(RLMSyncConfiguration *)syncConfiguration {
         @throw RLMException(@"Cannot set a sync configuration which has an errored-out user.");
     }
 
-    // Ensure sync manager is initialized, if it hasn't already been.
-    [RLMSyncManager sharedManager];
     NSAssert(user.identity, @"Cannot call this method on a user that doesn't have an identity.");
     self.config.in_memory = false;
-    self.config.sync_config = std::make_shared<realm::SyncConfig>([syncConfiguration rawConfiguration]);
+    self.config.sync_config = std::make_shared<realm::SyncConfig>(std::move([syncConfiguration rawConfiguration]));
     self.config.schema_mode = realm::SchemaMode::Additive;
 
     if (syncConfiguration.customFileURL) {
         self.config.path = syncConfiguration.customFileURL.path.UTF8String;
     } else {
         self.config.path = SyncManager::shared().path_for_realm(*[user _syncUser],
-                                                                self.config.sync_config->realm_url);
+                                                                [[user pathForPartitionValueHash:[@(self.config.sync_config->partition_value.c_str()) hash]] UTF8String]);
```

Comment:
`-[NSString hash]` is not a suitable hash function for this. There is no guarantee that it will remain stable between OS versions or even runs of an application.